presto-main/src/main/java/io/prestosql/connector/system/SystemTablesMetadata.java

```diff
@@ -156,12 +157,24 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
 
         TupleDomain<ColumnHandle> oldDomain = table.getConstraint();
         TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary());
+        if (oldDomain.equals(newDomain) && !constraint.predicate().isPresent()) {
+            return Optional.empty();
+        }
+
+        SystemTable systemTable = checkAndGetTable(session, table);
+        if (systemTable instanceof JdbcTable) {
```

Comment:
For the cursor method, we translate `TupleDomain<ColumnHandle>`, into ordinals:  `TupleDomain<Integer>`.

For `applyFilter` 
*  We can't pass `Constraint` since  it is expressed in terms of `ColumnHandle`, but `SystemColumnHandle` is not SPI. And we can't translate to ordinals, because of  https://github.com/prestosql/presto/commit/f148e5616add76cef83688908a884e4b4d14107f (of course, we can revert that, but I would strongly prefer this to be a follow up)
* for same reason we cannot have the method return `ConstraintApplicationResult`... The `applyFilter`  returns refined domain.

I concluded we are not ready to add this method to `SystemTable`. I considered adding a new interface for this, but decided it's not warranted in current situation.
Hence this is how it is. But this is definitely easy to improve in the future, when we conclude this is ready. (So far I didn't change SPI in any way, so no debt on this aspect.)

Reply:
Can you add a comment with a TODO so that future readers know *why* it was done this way?

Reply:
Added some comment in the code too.